### PR TITLE
feat(k8s): Support running cloudquery k8s plugin in k8s cluster using serviceaccount

### PR DESCRIPTION
**Pull request by lshmouse**
#### Summary

Support running cloudquery k8s plugin in k8s cluster using serviceaccount.

We are using argo workflow(https://argoproj.github.io/argo-workflows/) to schedule the cloudquery k8s resource syncing every day in the k8s pod.  Currently the client in k8s plugin only support kubeconfig env,  so we need the pack the kubeconfig into the docker image or put it in a configmap, which is not the k8s-way. 

So we want to add the support to create the client  in k8s plugin  using k8s serviceaccount.

The changes in the PR is a little hack. Suggestions are welcomed. Thanks~

### CODE REVIEW

File: plugins/source/k8s/client/client.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package client
 import (
 	"context"
 	"fmt"
+	"k8s.io/client-go/rest"
```
Comment (candiduslynx) [inline plugins/source/k8s/client/client.go:6]: please format with `goimports`

File: plugins/source/k8s/client/client.go [hunk 2]
```diff
@@ -133,6 +134,10 @@ func buildKubeClient(kubeConfig api.Config, ctx string) (*kubernetes.Clientset,
 		&clientcmd.ClientConfigLoadingRules{},
 	)
 	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		logger.Warn().Msg("Failed to create k8s client, fallback to use the in-cluster config")
+	}
+	restConfig, err = rest.InClusterConfig()
```
Comment (candiduslynx) [inline plugins/source/k8s/client/client.go:140]: you're always overwriting `restConfig`

File: plugins/source/k8s/config/cloudquery-k8s-cluster-role.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (candiduslynx) [inline plugins/source/k8s/config/cloudquery-k8s-cluster-role.yaml:1]: if that's a test file it should either be removed or it should be used in some test

**Review by candiduslynx — CHANGES_REQUESTED** [review]
Hi @lshmouse!
Thanks for thee PR.
I think it needs a bit more work (se the comments to the PR).

**Comment by lshmouse** [comment]
> Hi @lshmouse! Thanks for the PR. I think it needs a bit more work (se the comments to the PR).

Thanks for your reviewing, i will update the PR soon.

**Comment by lshmouse** [comment]
Update for @candiduslynx 's review. Thanks~

**Review by candiduslynx — APPROVED** [review]

**Comment by candiduslynx** [comment]
@lshmouse thank you for the contribution!

**Pull request merged**